File: Amazing Bird/Classes/ABSceneDelegate.m
//
//  ABSceneDelegate.m
//  Amazing Bird
//
//  Copyright (c) 2014 ChristianEnevoldsen. All rights reserved.
//

#import "ABSceneDelegate.h"
#import "ABAppDelegate.h"

@implementation ABSceneDelegate

- (void)scene:(UIScene *)scene willConnectToSession:(UISceneSession *)session options:(UISceneConnectionOptions *)connectionOptions
{
    if (![scene isKindOfClass:[UIWindowScene class]]) return;
    UIWindowScene *windowScene = (UIWindowScene *)scene;

    NSString *storyboardName = (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) ? @"Main_iPad" : @"Main_iPhone";
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:storyboardName bundle:nil];
    UIViewController *rootVC = [storyboard instantiateInitialViewController];

    self.window = [[UIWindow alloc] initWithWindowScene:windowScene];
    self.window.rootViewController = rootVC;
    [self.window makeKeyAndVisible];
}

- (void)sceneWillEnterForeground:(UIScene *)scene
{
    [(ABAppDelegate *)[UIApplication sharedApplication].delegate load];
}

- (void)sceneDidEnterBackground:(UIScene *)scene
{
    [(ABAppDelegate *)[UIApplication sharedApplication].delegate save];
}

@end

File: Amazing Bird/Classes/ABSceneDelegate.swift
import UIKit

class ABSceneDelegate: UIResponder, UIWindowSceneDelegate {
    var window: UIWindow?

    func scene(
        _ scene: UIScene,
        willConnectTo session: UISceneSession,
        options connectionOptions: UIScene.ConnectionOptions
    ) {
        guard let windowScene = scene as? UIWindowScene else { return }
        let storyboardName = UIDevice.current.userInterfaceIdiom == .pad ? "Main_iPad" : "Main_iPhone"
        let storyboard = UIStoryboard(name: storyboardName, bundle: nil)
        let rootVC = storyboard.instantiateInitialViewController()!
        window = UIWindow(windowScene: windowScene)
        window?.rootViewController = rootVC
        window?.makeKeyAndVisible()
    }

    func sceneWillEnterForeground(_ scene: UIScene) {
        ABPlayer.shared.load()
    }

    func sceneDidEnterBackground(_ scene: UIScene) {
        ABPlayer.shared.save()
    }
}
